src/pudl/glue/epacems_unitid_eia_plant_crosswalk.py

```diff
@@ -0,0 +1,119 @@
+EPA's github account at www.github.com/USEPA".
+"""
+import logging
+
+import pandas as pd
+
+import pudl
+from pudl.metadata.fields import apply_pudl_dtypes
+from pudl.workspace.datastore import Datastore
+
+logger = logging.getLogger(__name__)
+
+
+def extract(ds: Datastore) -> pd.DataFrame:
+    """Extract the EPACEMS-EIA Crosswalk from the Datastore."""
+    with ds.get_zipfile_resource(
+        "epacems_unitid_eia_plant_crosswalk",
+        name="epacems_unitid_eia_plant_crosswalk.zip",
+    ).open("camd-eia-crosswalk-master/epa_eia_crosswalk.csv") as f:
+        return pd.read_csv(f)
+
+
+def transform(
+    epa_eia_crosswalk: pd.DataFrame, generators_entity_eia: pd.DataFrame
+) -> dict[str, pd.DataFrame]:
+    """Clean up the EPACEMS-EIA Crosswalk file and split it into normalized tables.
+
+    Args:
+        epa_eia_crosswalk: The result of running this module's extract() function.
+        generators_entity_eia: The generators_entity_eia table.
+
+    Returns:
+        A dictionary of three normalized DataFrames comprised of the data
+        in the original crosswalk file. EPA plant id to EPA unit id; EPA plant
+        id to EIA plant id; and EIA plant id to EIA generator id to EPA unit
+        id. Includes no nan values.
+    """
+    logger.info("Cleaning up the epacems-eia crosswalk")
+
+    column_rename = {
+        "camd_unit_id": "unit_id_epa",
+        "camd_plant_id": "plant_id_epa",
+        "eia_plant_name": "plant_name_eia",
+        "eia_plant_id": "plant_id_eia",
+        "eia_generator_id": "generator_id",
+    }
+
+    # Basic column rename, selection, and dtype alignment.
+    crosswalk_clean = (
+        epa_eia_crosswalk.pipe(pudl.helpers.simplify_columns)
+        .rename(columns=column_rename)
+        .filter(list(column_rename.values()))
+        .pipe(apply_pudl_dtypes, "eia")
+    )
+
+    # The crosswalk is a static file: there is no year field. The plant_id_eia and
+    # generator_id fields, however, are foreign keys from an annualized table. If the
+    # fast ETL is run (on one year of data) the test will break because the crosswalk
+    # tables with plant_id_eia and generator_id contain values from various years. To
+    # keep the crosswalk in alignment with the available eia data, we'll restrict it
+    # based on the generator entity table which has plant id and generator id.
+    crosswalk_clean = crosswalk_clean.merge(
+        generators_entity_eia[["plant_id_eia", "generator_id"]],
+        on=["plant_id_eia", "generator_id"],
+        how="inner",
+    )
+
+    # There are some eia generator_id values in the crosswalk that don't match the eia
+    # generator_id values in the generators_eia860 table where the foreign keys are
+    # stored. All of them appear to have preceeding zeros. I.e.: 0010 should be 10.
+    # This makes sure to nix preceeding zeros on crosswalk generator ids that are all
+    # numeric. I.e.: 00A10 will stay 00A10 but 0010 will become 10.
+    crosswalk_clean.loc[
```

Comment:
Take a look at `pudl.helpers.fix_leading_zero_gen_ids()` -- that's the function we're already using to do this kind of cleanup on the generator IDs in the EIA data itself (which contains a mix of leading zeros, or not, for the same generators... which is crazymaking). If possible it would be good to apply exactly the same fixes to both datasets so we can be sure that they match up.